interoperator/pkg/controller/multiclusterdeploy/provisioner/provisioner_controller.go

```diff
@@ -0,0 +1,433 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("provisioner.controller")
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new SFDefaultScheduler Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+// USER ACTION REQUIRED: update cmd/manager/main.go to call this osb.Add(mgr) to install this Controller
+func Add(mgr manager.Manager) error {
+	clusterRegistry, err := registry.New(mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	provisionerMgr, err := provisioner.New(mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	err = provisionerMgr.FetchStatefulset()
+	if err != nil {
+		return err
+	}
+
+	return add(mgr, newReconciler(mgr, clusterRegistry, provisionerMgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, clusterRegistry registry.ClusterRegistry, provisionerMgr provisioner.Provisioner) reconcile.Reconciler {
+	return &ReconcileProvisioner{
+		Client:          mgr.GetClient(),
+		scheme:          mgr.GetScheme(),
+		clusterRegistry: clusterRegistry,
+		provisioner:     provisionerMgr,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	cfgManager, err := config.New(mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	interoperatorCfg := cfgManager.GetConfig()
+	// Create a new controller
+	c, err := controller.New("provisioner-controller", mgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: interoperatorCfg.ProvisionerWorkerCount,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to SFDefaultScheduler
+	err = c.Watch(&source.Kind{Type: &resourcev1alpha1.SFCluster{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileProvisioner{}
+
+// ReconcileProvisioner reconciles a SFDefaultScheduler object
+type ReconcileProvisioner struct {
+	client.Client
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+	provisioner     provisioner.Provisioner
+}
+
+// Reconcile reads that state of the cluster for a SFDefaultScheduler object and makes changes based on the state read
```

Comment:
Do the first few lines of the comments make sense ?

Reply:
Changed